File: julia/day1.jl
module day1
	# part 1
	input = split(read("data/day1.txt", String))

  input_modified = map(x -> parse(Int64, x), input)

	part1 = sum(input_modified)
	println("Solution to Part 1: $part1")

	# part2
	function part2(input = input_modified)
    current = 0
    repeating_freqs = Set{Int64}([0])

    while true
      for x in input_modified
        current += x
        if current in repeating_freqs
          return(current)
        end
        push!(repeating_freqs, current)
      end
    end
  end
  part2_solution = part2()
  println("Solution to part2: $part2_solution")
end

File: python/day1.py
lines = [int(x.strip()) for x in open("data/day1.txt")]

def part1(lines = lines):
  return(sum(lines))

def part2(lines = lines):
  repeating = set([0])
  current = 0
  while 1:
    for x in lines:
      current += x
      # print(current)
      # print(repeating)
      if current in repeating:
        return(current)
      else:
        repeating.add(current)

print(part1(lines))
print(part2(lines))
  
